src/main/scala/sigmastate/utils/ByteReader.scala

```diff
@@ -27,8 +55,77 @@ class ByteBufferReader(buf: ByteBuffer) extends ByteReader {
   @inline override def getByte(): Byte = buf.get
   @inline override def getUByte(): Int = buf.get & 0xFF
   @inline override def getShort(): Short = buf.getShort()
-  @inline override def getInt(): Int = buf.getInt()
-  @inline override def getLong(): Long = buf.getLong()
+
+  /**
+    * Decode signed Int previously encoded with [[ByteArrayWriter.putInt]] using VLQ with ZigZag.
+    *
+    * @note Uses ZigZag encoding. Should be used to decode '''only''' a value that was previously
+    *       encoded with [[ByteArrayWriter.putInt]].
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @return signed Int
+    */
+  @inline override def getInt(): Int =
+  // should only be changed simultaneously with `putInt`
+    getSInt()
+
+  /**
+    * Decode signed Int previously encoded with [[ByteArrayWriter.putSInt]] using VLQ with ZigZag.
+    *
+    * @note Uses ZigZag encoding. Should be used to decode '''only''' a value that was previously
+    *       encoded with [[ByteArrayWriter.putSInt]].
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @return signed Int
+    */
+  @inline def getSInt(): Int = ByteBufferReader.decodeZigZagInt(getULong().toInt)
```

Comment:
Maybe getInt? SInt is a type, so the reading may be confusing. Also Int, UInt is typical pair.

Reply:
@ergomorphic Agreed. Not much sense in having both `*SInt` and `*Int` dangling around when  `*Int` behavior is settled (VLQ + ZigZag vs. VLQ).

Reply:
Fixed. And `*Long` as well`